Watchman.Tests/CloudFormationDeploymentTests.cs

```diff
@@ -135,6 +135,146 @@ await secondTestContext
+                new AlertingGroupServices());
+
+            var secondTestContext = new TestingIocBootstrapper()
+                .WithCloudFormation(cloudformation.Instance)
+                .WithConfig(emptyConfig);
+
+
+            await secondTestContext
+                .Get<AlarmLoaderAndGenerator>()
+                .LoadAndGenerateAlarms(RunMode.GenerateAlarms);
+
+            stack = cloudformation
+                .Stack("Watchman-test");
+
+            //check we deployed an empty stack and deleted the redundant resources
+            Assert.That(stack, Is.Not.Null);
+            Assert.That(stack.Resources.Any(x => x.Value.Type == "AWS::CloudWatch::Alarm"), Is.False);
+        }
+
+        [Test]
+        public async Task TestNullRefIssueWhenMixingOldAndNewConfig()
+        {
+            var config = new WatchmanConfiguration()
+            {
+                AlertingGroups = new List<AlertingGroup>()
+                {
+                    new AlertingGroup()
+                    {
+                        Name = "ag1",
+                        AlarmNameSuffix = "ag1suffix",
+                        Services = null,
+                        DynamoDb = new DynamoDb()
+                        {
+                            Tables = new List<Table>()
+                            {
+                                new Table()
+                                {
+                                    Name = "banana"
+                                }
+                            }
+                        }
+                    },
+                    new AlertingGroup()
+                    {
+                        Name = "ag2",
+                        AlarmNameSuffix = "ag2suffix",
+
+                        Services = new AlertingGroupServices()
+                        {
+                            AutoScaling = new AwsServiceAlarms<AutoScalingResourceConfig>()
+                            {
+                                Resources = new List<ResourceThresholds<AutoScalingResourceConfig>>()
+                                {
+                                    new ResourceThresholds<AutoScalingResourceConfig>()
+                                    {
+                                        Name = "asg"
+                                    }
+                                }
+                            }
+                        }
+                    }
+
+                }
+            };
+
+            var cloudformation = new FakeCloudFormation();
+
+            var context = new TestingIocBootstrapper()
+                .WithCloudFormation(cloudformation.Instance)
+                .WithConfig(config);
+
+            context.GetMock<IAmazonAutoScaling>()
+                .HasAutoScalingGroups(new AutoScalingGroup[0]);
+
+            Exception caught = null;
+            
+            try 
+            {
+                await context.Get<AlarmLoaderAndGenerator>()
```

Comment:
Why you don't you use `Assert.DoesNotThrowAsync();` ?

Reply:
because I saw some discussion on xunit about why they removed it. (I know we're using nunit but it seemed to make sense as being unnecessary). 
I was also just thinking of leaving it as is, or adding a comment, or adding `Assert.Pass()` explicitly at the end. It all seems a bit unnecessary as the test will fail anyway if it throws

Reply:
Fair enough. We can leave it as it is then.

Reply:
Comment works too. Just saw it.